setup.py

```diff
@@ -666,7 +671,40 @@ def prefix(directory, files):
 
 # compression libs
 
-def get_charls_clib(field=None):
+
+def _get_bz2_clib(field=None):
+    """BZip2 static lib build config"""
+    bzip2_dir = "src/bzip2"
+
+    config = dict(
+        sources=prefix(bzip2_dir, [
+            "blocksort.c",
+            "huffman.c",
+            "crctable.c",
+            "randtable.c",
+            "compress.c",
+            "decompress.c",
+            "bzlib.c",
+        ]),
+        include_dirs=[bzip2_dir],
+        macros=[],
+        cflags=[
+            "-Wall",
+            "-Winline",
+            "-O2",
+            "-g",
+            "-D_FILE_OFFSET_BITS=64"
+        ],
+    )
+
+    if field is None:
+        return config
+    if field in ('extra_link_args', 'libraries'):
+        return []
```

Comment:
'side question' not sure I get this part. Why is this returning a list ? I would expect an empty dict.

Reply:
If `field` is `None`, it returns a `dict` which contains the build config for the static lib which is passed to `BuildCLib` setuptools command.
If `field` is provided, it returns the value of the field. It is called with way from the hdf5 plugin code using the compression lib to get what needs to be passed for linking with the compression lib to the setuptools `Extension`. `Extension` expects `list`s for 'extra_link_args' and 'libraries'.

It cannot be all stored in a `dict` since 'extra_link_args' and 'libraries' are not expected by `BuildCLib`.
It would probably be best to use a class/dataclass/namedtuple here, or a function to extract what's relevant to `BuildCLib`. Here I've minimised the rewrite, but I agree it makes things complex